contracts/Registry.sol

```diff
@@ -101,21 +104,24 @@ contract Registry is Ownable, FundsRecovery {
         }
     }
 
-    function registerAccountant(address _accountantOperator, uint256 _stakeAmount, uint16 _accountantFee) public {
-        require(_accountantOperator != address(0), "accountant cannot be registered on zero address");
+    function registerAccountant(address _accountantOperator, uint256 _stakeAmount, uint16 _accountantFee, uint256 _maxLoan) public {
+        require(_accountantOperator != address(0), "operator can't be zero address");
         require(_stakeAmount >= minimalAccountantStake, "accountant have to stake at least minimal stake amount");
 
         address _accountantId = getAccountantAddress(_accountantOperator);
         require(!isAccountant(_accountantId), "accountant already registered");
 
-        token.transferFrom(msg.sender, address(this), _stakeAmount);
-        totalStaked = totalStaked.add(_stakeAmount);
-
         // Deploy accountant contract (mini proxy which is pointing to implementation)
         AccountantContract _accountant = AccountantContract(deployMiniProxy(uint256(_accountantOperator), accountantImplementation));
-        _accountant.initialize(address(token), _accountantOperator, _accountantFee);
 
-        accountants[address(_accountant)] = Accountant(_accountantOperator, _stakeAmount);
+        // Transfer stake into accountant smart contract
+        token.transferFrom(msg.sender, address(_accountant), _stakeAmount);
+
+        // Initialise accountant 
+        _accountant.initialize(address(token), _accountantOperator, _accountantFee, _maxLoan);
```

Comment:
It's way for accountant to protect himself from not wanted punishment.

Imagine situation when someone will make loan for his channel 100 000 tokens. Then he will not settle for a while but will get a lot of payments. Then he can wait until accountant available balance is less than 100 000 and just try to rebalance. This will enable punishment mode. To refill his channel back, accountant will need to settle a lot of small consumer promises. 

And yes, there could be default stake of 10 000 and channel of 100 000. With current punishment logic, we will drain whole stake in 10 days.